Sources/xcproj/PBXProj.swift

```diff
@@ -18,34 +17,38 @@ public class PBXProj: Decodable {
     /// Project root object.
     public var rootObject: String
 
-    public var buildFiles: [PBXBuildFile] = []
-    public var aggregateTargets: [PBXAggregateTarget] = []
-    public var containerItemProxies: [PBXContainerItemProxy] = []
-    public var copyFilesBuildPhases: [PBXCopyFilesBuildPhase] = []
-    public var groups: [PBXGroup] = []
-    public var fileElements: [PBXFileElement] = []
-    public var configurationLists: [XCConfigurationList] = []
-    public var versionGroups: [XCVersionGroup] = []
-    public var buildConfigurations: [XCBuildConfiguration] = []
-    public var variantGroups: [PBXVariantGroup] = []
-    public var targetDependencies: [PBXTargetDependency] = []
-    public var sourcesBuildPhases: [PBXSourcesBuildPhase] = []
-    public var shellScriptBuildPhases: [PBXShellScriptBuildPhase] = []
-    public var resourcesBuildPhases: [PBXResourcesBuildPhase] = []
-    public var frameworksBuildPhases: [PBXFrameworksBuildPhase] = []
-    public var headersBuildPhases: [PBXHeadersBuildPhase] = []
-    public var nativeTargets: [PBXNativeTarget] = []
-    public var fileReferences: [PBXFileReference] = []
-    public var projects: [PBXProject] = []
-    public var referenceProxies: [PBXReferenceProxy] = []
```

Comment:
With those changes, we'd need to bump the major version again because those are breaking changes. What do you think about having something like this:

```swift
@available(*, deprecated, message: "Use referenceProxiesCollection instead") public var referenceProxies: [PBXReferenceProxy] {
  return referenceProxiesCollection.values
}
public var referenceProxiesCollection: ReferenceableCollection<PBXReferenceProxy> = ReferenceableCollection<PBXReferenceProxy>()
```
I'm not quite happy about the name though. I think we could avoid having to append `Collection` by using a class a a namespace:

```swift
class PBXProj {
  class PBXProjObjects {
    public var referenceProxies: ReferenceableCollection<PBXReferenceProxy> = ReferenceableCollection<PBXReferenceProxy>()
  }
  var objects: PBXProjObjects = PBXProjObjects()
}
```
You could access them as `proj.objects. referenceProxies` instead of `proj. referenceProxies` that would be deprecated.

What are your thoughts on this @rahul-malik  / @yonaskolb ?

Reply:
Yeah, I quite like the idea of encapsulating all the collections into one objects class. Then it keeps all those huge switch statements in one place. 
As above though I still think those actual collections should just be dictionaries instead of a `ReferenceableCollection`